docs/tasks/debug-application-cluster/core-metrics-pipeline.md

```diff
@@ -0,0 +1,47 @@
+---
+approvers:
+- fgrzadkowski
+- piosz
+title: Core metrics pipeline
+---
+
+Resource usage metrics are important for:
+
+- Kubernetes components to operate (e.g. autoscalers to automatically scale user applications)
+- Kubernetes users to understand their applications behavior
+
+In Kubernetes they are available through Metrics API.
+
+## Metrics API
+
+Metrics API provides current resource usage metrics for pods and nodes. It doesn't offer historical metrics, so one needs to
+archive metrics themselves. In the future Historical API might be defined but it will not be implemented as a part of
+the core metrics pipeline.
+
+The API is discoverable through the same endpoint as the other Kubernetes APIs under `/apis/metrics.k8s.io/` path.
+It also offers the same security, scalability and reliability guarantees.
+
+The API is defined in [k8s.io/metrics](https://github.com/kubernetes/metrics/tree/master/pkg/apis/metrics) repository,
+where the detailed semantic is explained.
+
+### Supported endpoints
+
+The list of supported endpoints in the API:
+
+- `/nodes` - all node metrics; support for `labelSelector` parameter
+- `/nodes/{node}` - metrics for the specified node
+- `/namespaces/{namespace}/pods` - all pod metrics within the namespace; support for `labelSelector` parameter
+- `/namespaces/{namespace}/pods/{pod}` - metrics for the specified pods
+
+## Metrics Server
+
+[Metrics Server](https://github.com/kubernetes-incubator/metrics-server) is a cluster-wide aggregator of monitoring data.
+Starting from Kubernetes 1.8 it's work on all Kubernetes setups
```

Comment:
Starting from Kubernetes 1.8 it's __deployed by default in__ all Kubernetes setups

Is it really? What about e.g. kopf?

Reply:
Would it be useful to include steps for the user to deploy the metrics server themselves, if it is not deployed by default (for ex on pre-1.8 cluster)?
Also how can a user verify/debug that it is running? Should they run `kubectl get deployments` and expect to see a specific deployment?

Reply:
Agreed. A complete story would be more useful.

Reply:
Improved the section